### PR TITLE
LIBTD-1424: HTML Sanitizer on Input Fields Prior to Entering Data to DB or Fedora

### PR DESCRIPTION
**Pull request by shabububu**
**JIRA Ticket**: https://webapps.es.vt.edu/jira/browse/LIBTD-1424

# What does this Pull Request do?
For this PR, I've added tinymce for the form input on a work's description, program notes, and technical specs. The user profile's personal statement and collection's description has been modified as well. This is one step to add "due diligence" security to COMPEL, while also allowing the user some freedom to stylize their works and profile.

See the Jira ticket for more information.

# What are the changes?
For this PR, I've added tinymce for the form input on a work's description, program notes, and technical specs. The user profile's personal statement and collection's description has been modified as well. I carried over and modified the `_head_tag_content.html.erb` from the Hyrax 2.1.0 gem to allow for using tinymce without administrative privileges.

# How should this be tested?

One way of testing this out would be to create both a Performance work and Composition work, adding stylized content for the modified fields.   Also, attempt modifying your user's personal statement. Finally, create a collection with a description. If the styles come out as expected and potentially malicious code is not allowed (e.g. <script>...</script>), then it should be working as expected. You may need to use the rails and fedora consoles to see if they are actually stored as expected. [Edit: One clarification, the potentially malicious tags should still be visible, but not rendered as html]

# Additional Notes:
* I'm not a security expert, but this seemed like one approach to adding security without detracting from user experience. This should be considered while reviewing this PR.
* Use the LIBTD-1424 branch for testing

# Interested parties
@whunter @tingtingjh

### CODE REVIEW

**Comment by shabububu** [comment]
Based on slack conversation with @tingtingjh , I'm adding a new attribute renderer for Technical Specs and Program Notes.

File: app/views/hyrax/base/_attribute_rows.html.erb [hunk 1]
```diff
@@ -8,7 +8,7 @@
 <%= presenter.attribute_to_html(:resource_type, render_as: :faceted, html_dl: true) %>
 <%= presenter.attribute_to_html(:medium, html_dl: true) %>
 <%= presenter.attribute_to_html(:source, render_as: :external_link, html_dl: true) %>
-<%= presenter.attribute_to_html(:technical_specs, html_dl: true) %>
-<%= presenter.attribute_to_html(:program_notes, html_dl: true) %>
+<%= presenter.attribute_to_html(:technical_specs, render_as: :markdown_generated) %>
```
Comment (tingtingjh) [inline app/views/hyrax/base/_attribute_rows.html.erb]: Should the "html_dl: true" be kept still?

**Review by tingtingjh — CHANGES_REQUESTED** [review]
For multi_value description field in works, the links are not rendered correctly (not clickable). You can try to fix this by modifying the records/show_fields/_description.html.erb.

File: app/views/hyrax/base/_attribute_rows.html.erb (hunk 1 above)
Comment (shabububu) [reply app/views/hyrax/base/_attribute_rows.html.erb]: That makes sense. If all the other ones use it, then we should keep it in for consistency. I'll add the `html_dl: true back` in.

**Comment by shabububu** [comment]
My original intention was to not allow clickable links for the fields that I modified to use tinymce.  ~~So, in order to make the works description more in line with the technical specs and program notes, I'm planning on modifying the helper function `link_to_html` to not show links by default. [Edit: Just remove the whole ability to create links and iconify them.]~~ Maybe we can keep links in and have the `autolink` gem do some sanitizing of them. I need to think more on this.

After testing some more, there's also another issue related to tinymce for the case of description. It's the only one right now that is multi-value. When attempting to enter multiple descriptions, the UI doesn't seem to respond like it should. For example, when I try to enter more than one description, the UI doesn't always seem to enable the text area for the newly created one. I'm still investigating why.

**Comment by shabububu** [comment]
I just pushed changes so that the following can be expected from the form input. Basically, I'm using the helper `link_to_html`, which in turn calls the `auto_link` gem to do some sanitization at rendering time.

| Input from TinyMCE | Expected output at rendering |
| -------------------- | ----------------------------- |
| `www.google.com`      | clickable link rendered           |
| `<script> alert()</script>` | tags viewable but not rendered|
| `javascript:alert(‘bad’)` | tags viewable but not rendered |
| `<a href="http://www.google.com">google.com</a>` | tags viewable but not rendered, BUT clickable link rendered. This is potentially confusing, but I think the user will get it. |
| `Visit http://www.google.com/ or e-mail me@example.com` | two clickable links rendered around the links and email|

I'm still working on the UI issue when trying to enter multiple descriptions.

**Comment by shabububu** [comment]
@tingtingjh  Okay, I probably would benefit from another set of eyes to get the additional tinymce fields to accept input properly (the UI issue). What I believe is the problem is that we need to do a javascript call:
```
tinyMCE.init({
      selector: 'textarea.tinymce'
 });
```
after the new description block has been added to the form after clicking "+Add another abstract or summary". I'm going to keep working on this, but I think I've been staring at this too long and would benefit from another set of eyes. 

What I'm not sure about is when to call that tinyMCE.init. Ideally, I'd call it once the new textarea has been added. However, I'm not sure how to do that.

The current functionality has multiple tinymce dom nodes being added, but with non-unique ids.

**Comment by shabububu** [comment]
@tingtingjh and I tried different things to get the tinymce and multi-value description to work, but we weren't able to get it done in time. For this PR, I'm going to back out of using tinymce for the work description (i.e., remove `app/views/records/edit_fields/_description.html.erb`). We hope that we can get it working at a later time.

**Review by tingtingjh — APPROVED** [review]

**Pull request merged**